modules/reactor/src/meshgenerators/PolygonMeshGeneratorBase.C

```diff
@@ -0,0 +1,1229 @@
+  Real bin_radial_distance;
+  for (unsigned int l = 0; l < rings.size(); l++)
+  {
+    if (l == 0)
+      pin_radius_interval_length[l] =
+          ring_radii[l] / rings[l]; // the pin radius interval for each ring_radii/subdomain
+    else
+      pin_radius_interval_length[l] =
+          (ring_radii[l] - ring_radii[l - 1]) /
+          rings[l]; // the pin radius interval for each ring_radii/subdomain
+
+    // add rings in each pin subdomain
+    for (unsigned int k = 0; k < rings[l]; k++)
+    {
+      (*node_id)++;
+      if (l == 0)
+        bin_radial_distance =
+            ((k + 1) * pin_radius_interval_length[l]); // this is from the cell/pin center to
+                                                       // the first circle
+      else
+        bin_radial_distance = (ring_radii[l - 1] + (k + 1) * pin_radius_interval_length[l]);
+      const Real pin_corner_p_x = corner_p[0][0] * bin_radial_distance / (0.5 * corner_to_corner);
+      const Real pin_corner_p_y = corner_p[0][1] * bin_radial_distance / (0.5 * corner_to_corner);
+
+      // pin_corner_p(s) are the points in the pin region, on the bins towards the six corners,
+      // at differnt intervals
+      Point p = Point(pin_corner_p_x, pin_corner_p_y, 0.0);
+      (*nodes)[*node_id] = mesh->add_point(p, *node_id);
+
+      for (unsigned int j = 1; j <= angle_number; j++)
+      {
+        (*node_id)++;
+        const Real cell_boundary_p_x =
+            corner_p[0][0] + (corner_p[1][0] - corner_p[0][0]) *
+                                 (azimuthal_tangent.size() == 0 ? ((Real)j / (Real)angle_number)
+                                                                : (azimuthal_tangent[j] / 2.0));
+        const Real cell_boundary_p_y =
+            corner_p[0][1] + (corner_p[1][1] - corner_p[0][1]) *
+                                 (azimuthal_tangent.size() == 0 ? ((Real)j / (Real)angle_number)
+                                                                : (azimuthal_tangent[j] / 2.0));
+        // cell_boundary_p(s) are the points on the cell's six boundries (flat sides) at
+        // differnt azimuthal angles
+        const Real pin_azimuthal_p_x =
+            cell_boundary_p_x * bin_radial_distance /
+            std::sqrt(Utility::pow<2>(cell_boundary_p_x) + Utility::pow<2>(cell_boundary_p_y));
+        const Real pin_azimuthal_p_y =
+            cell_boundary_p_y * bin_radial_distance /
+            std::sqrt(Utility::pow<2>(cell_boundary_p_x) + Utility::pow<2>(cell_boundary_p_y));
+
+        // pin_azimuthal_p are the points on the bins towards differnt azimuthal angles, at
+        // differnt intervals; excluding the ones produced by pin_corner_p
+        Point p = Point(pin_azimuthal_p_x, pin_azimuthal_p_y, 0.0);
+        (*nodes)[*node_id] = mesh->add_point(p, *node_id);
+      }
+    }
+  }
+  return dynamic_pointer_cast<ReplicatedMesh>(mesh);
+}
+
+// 'background_nodes' creates nodes for the ring-to-polygon transition region (i.e., background) of
```

Comment:
can we think of a better name than background?
or is this the standard name in sfr terminology

Reply:
The central ring regions usually represent pins in a reactor, while the transition regions are reactor matrix (solid or liquid). So `background` seems to be a reasonable name.